admin/class-admin-recommended-replace-vars.php

```diff
@@ -0,0 +1,161 @@
+		// Settings - special pages.
+		'search'                  => array( 'sitename', 'searchphrase', 'sep', 'page' ),
+		'404'                     => array( 'sitename', 'sep' ),
+	);
+
+	/**
+	 * Determines the page type of the current term.
+	 *
+	 * @param string $taxonomy The taxonomy name.
+	 *
+	 * @return string The page type.
+	 */
+	public function determine_for_term( $taxonomy ) {
+		$supported_taxonomies = array( 'category', 'tag', 'post_format' );
+
+		// Add WooCommerce specific types.
+		if ( $this->is_woocommerce_active() ) {
+			$supported_taxonomies[] = 'product_cat';
+			$supported_taxonomies[] = 'product_tag';
+			$supported_taxonomies[] = 'product_brand';
+			$supported_taxonomies[] = 'product_shipping_class';
+			// This is the for the plugin: Perfect WooCommerce Brands.
+			$supported_taxonomies[] = 'pwb-brand';
+		}
+
+		if ( in_array( $taxonomy, $supported_taxonomies, true ) ) {
+			return $taxonomy;
+		}
+
+		return 'term-in-custom-taxomomy';
+	}
+
+	/**
+	 * Determines the page type of the current post.
+	 *
+	 * @param WP_Post $post The WordPress global post variable.
+	 *
+	 * @return string The page type.
+	 */
+	public function determine_for_post( $post ) {
+		if ( $post instanceof WP_Post === false ) {
+			return 'post';
+		}
+
+		if ( $post->post_type === 'page' && $this->is_homepage( $post ) ) {
+			return 'homepage';
+		}
+
+		$supported_post_types = array( 'post', 'page' );
+
+		// Add WooCommerce specific types.
+		if ( $this->is_woocommerce_active() ) {
+			$supported_post_types[] = 'product';
+		}
+
+		if ( in_array( $post->post_type, $supported_post_types, true ) ) {
+			return $post->post_type;
+		}
+
+		return 'custom_post_type';
+	}
+
+	/**
+	 * Retrieves the recommended replacement variables for the given post type.
+	 *
+	 * @param string $post_type The post type.
+	 *
+	 * @return array The recommended replacement variables.
+	 */
+	public function get_recommended_replacevars_for( $post_type ) {
+		if ( ! isset( $this->recommended_replace_vars[ $post_type ] ) ) {
+			return array();
+		}
+
+		return $this->recommended_replace_vars[ $post_type ];
+	}
+
+	/**
+	 * Retrieves the recommended replacement variables.
+	 *
+	 * @return array The recommended replacement variables.
+	 */
+	public function get_recommended_replacevars() {
+		return $this->recommended_replace_vars;
+	}
+
+	/**
+	 * Determines whether or not a post is the homepage.
+	 *
+	 * @param WP_Post $post The WordPress global post variable.
+	 *
+	 * @return bool True if the given post is the homepage.
+	 */
+	private function is_homepage( $post ) {
+		if ( $post instanceof WP_Post === false ) {
+			return false;
+		}
+
+		/*
+		 * The page on front returns a string with normal WordPress interaction, while the post ID is an int.
+		 * This way we make sure we always compare strings.
+		 */
+		$post_id_string = '' . $post->ID;
+		$page_on_front = '' . get_option( 'page_on_front' );
```

Comment:
It is nice to do `( string )`

But you are doing a comparison of ID's. I would choose to cast them to `int`